src/container_registry/utilities.rs

```diff
@@ -0,0 +1,74 @@
+use crate::cmd;
+use crate::container_registry::Kind;
+use crate::error::{SimpleError, SimpleErrorKind};
+use retry::delay::Fibonacci;
+use retry::Error::Operation;
+use retry::OperationResult;
+
+pub fn docker_tag_and_push_image(
+    container_registry_kind: Kind,
+    docker_envs: Option<Vec<(&str, &str)>>,
+    image_name: String,
+    image_tag: String,
+    dest: String,
+) -> Result<(), SimpleError> {
+    let image_with_tag = format!("{}:{}", image_name, image_tag);
+    let registry_provider = match container_registry_kind {
+        Kind::DockerHub => "DockerHub",
+        Kind::Ecr => "AWS ECR",
+        Kind::Docr => "DigitalOcean Registry",
+    };
+    let docker_environment_variables = match docker_envs {
+        None => vec![("", "")],
```

Comment:
why not just vec![] ?

Reply:
useless, therefore essential :)

You're right, removing them